src/build/__init__.py

```diff
@@ -20,7 +20,17 @@
 from typing import AbstractSet, Any, Callable, Dict, Iterator, Mapping, Optional, Sequence, Set, Tuple, Type, Union
 
 import pep517.wrappers
-import toml
+
+
+try:
+    import tomli as toml
+
+    TOMLDecodeError: Type[Exception] = toml.TOMLDecodeError
+except ModuleNotFoundError:  # pragma: no cover
+    import toml as _toml
+
+    toml = _toml  # type: ignore
+    TOMLDecodeError = _toml.TomlDecodeError
```

Comment:
```suggestion
try:
    from tomli import loads as toml_loads
    from tomli import TOMLDecodeError
except ImportError:  # pragma: no cover
    from toml import loads as toml_loads
    from toml import TomlDecodeError as TOMLDecodeError
```

Something like this plus changing these lines below
```python
with open(spec_file, encoding='UTF-8') as f:
    spec = toml.load(f)
```
to
```python
with open(spec_file, 'rb') as f:
        spec = toml_loads(f.read().decode())
```
will
1. avoid a [deprecation warning](https://github.com/hukkin/tomli/blob/6c3ae6b0836d4af2d3b434b594312d78af0b116a/tomli/_parser.py#L71) in tomli
2. avoid making illegal line ending conversions when opening the file

Reply:
I generally prefer to keep full imports, but since we were already doing it for the error, we can have `loads` too.